### PR TITLE
Carry 1510

### PR DESCRIPTION
**Pull request by moxiegirl**
Closes #1510 

ping @vieux @abronan

### CODE REVIEW

File: docs/discovery.md [hunk 1]
```diff
@@ -11,218 +11,196 @@ weight=4
...
+  **ZooKeeper**:
 
+        swarm join --advertise=<node_ip:2375> zk://<zookeeper_addr1>,<zookeeper_addr2>/<optional path prefix>
 
-Then start the Swarm manager on any machine.
+2. Start the Swarm manager on any machine or your laptop.
 
-    swarm manage -H tcp://<swarm_ip:swarm_port> file:///tmp/my_cluster
+  **Etcd**:
+        swarm manage -H tcp://<swarm_ip:swarm_port> etcd://<etcd_addr1>,<etcd_addr2>/<optional path prefix>
```
Comment (abronan) [inline docs/discovery.md]: markdown formatting seems to be broken for the code portion (maybe enforcing ``` ?)

File: docs/discovery.md [hunk 2]
```diff
@@ -11,218 +11,196 @@ weight=4
...
+        ...
 
-    $ swarm list file:///tmp/my_cluster
-    <node_ip1:2375>
-    <node_ip2:2375>
-    <node_ip3:2375>
+5. Try listing the nodes in your cluster.
 
+  **Etcd**:
+        swarm list etcd://<etcd_addr1>,<etcd_addr2>/<optional path prefix>
```
Comment (abronan) [inline docs/discovery.md]: Here as well

**Comment by moxiegirl** [comment]
@abronan sorted.

File: docs/discovery.md [hunk 3]
```diff
@@ -11,218 +11,201 @@ weight=4
...
-You use a hosted discovery service with Docker Swarm. The service
-maintains a list of IPs in your swarm. There are several available
-services, such as `etcd`, `consul` and `zookeeper` depending on what
-is best suited for your environment. You can even use a static
-file. Docker Hub also provides a hosted discovery service which you
-can use.
+## Using a distributed key/value store
 
-### Hosted Discovery with Docker Hub
+The recommend way to do node discovery in Swarm is Docker's libkv project. The libkv project is an abstraction layer over existing distributed key/value stores.  As of this writing, the project supports:
```
Comment (abronan) [inline docs/discovery.md]: recommend -> recommended (sorry missed that one :cry:)

**Comment by abronan** [comment]
Last small typo but LGTM when sorted :)

**Comment by moxiegirl** [comment]
Done. Thank you @abronan @vieux for a check and merge.

**Comment by vieux** [comment]
LGTM

**Pull request merged**